case-lib/lib.sh

```diff
@@ -35,7 +35,8 @@ fi
 
 # setup SOFCARD id
 if [ ! "$SOFCARD" ]; then
-    SOFCARD=$(grep -v 'sof-probes' /proc/asound/cards | grep 'sof-[a-z]' | awk '{print $1;}')
+	# remove leading spaces and check if the line starts with a number
+	SOFCARD=$(grep -v 'sof-probes' /proc/asound/cards | grep 'sof-[a-z]' | awk '{$1=$1};1 && $1 ~ /^[0-9]+$/ {print $1;}')
```

Comment:
OK, this is not what I expected. I understand `{$1=$1};1` now but it's very cryptic with 2 actions where the second one is implicit. I'm not even sure whether the current line has 2 or 3 actions in total, that's quite confusing. I think you can do the same with a single `{ }`action (untested)
```shell
# $1=$1 strips whitespace
SOFCARD=$(grep -v 'sof-probes' /proc/asound/cards |
   awk '/sof-[a-z]/ && $1 ~ /^[0-9]+$/ { $1=$1; print $1;} ' 
```
```

Reply:
# $1=$1 strips whitespace
SOFCARD=$(grep -v 'sof-probes' /proc/asound/cards |
   awk '/sof-[a-z]/ && $1 ~ /^[0-9]+$/ { $1=$1; print $1;}) '  works!

Reply:
> grep -v 'sof-probes' /proc/asound/cards | grep 'sof-[a-z]' | grep '^[0-9]' | awk '{print $1;}'

This does not make sure $1 is a number and does not strip whitespace (BTW `grep -e 'sof-[a-z]' -e '^[0-9]` would as clear and shorter and faster)

@fredoh9 `'/sof-[a-z]/ && $1 ~ /^[0-9]+$/ { $1=$1; print $1;}')` is not much more complex than the https://www.gnu.org/software/gawk/manual/html_node/Very-Simple.html example, unlike previous versions of this PR it has the same basic `expression { action }` structure documented in https://www.gnu.org/software/gawk/manual/html_node/Getting-Started.html: 

While it's often possible to read some Python or shell code without even knowing either, I agree that reading even basic `awk` syntax requires a little bit more upfront knowledge. If you think `expression { action }` is too complicated then you're basically saying not to use awk all - then what would be the alternative in this particular case?